plugins/plugin-github/che-plugin-github-ext-github/src/main/java/org/eclipse/che/ide/ext/github/client/GitHubClientServiceImpl.java

```diff
@@ -139,9 +146,9 @@ public void getRepositoriesByUser(String userName, @NotNull AsyncRequestCallback
 
     /** {@inheritDoc} */
     @Override
-    public void getAllRepositories(@NotNull AsyncRequestCallback<Map<String, List<GitHubRepository>>> callback) {
+    public Promise<List<GitHubRepository>> getAllRepositories() {
         String url = baseUrl + LIST_ALL;
-        asyncRequestFactory.createGetRequest(url).loader(loader).send(callback);
+        return asyncRequestFactory.createGetRequest(url).loader(loader).send(new GitHubAllRepositoriesUnmarshaller(dtoFactory));
```

Comment:
I think you can use DtoUnmarshallerFactory.newArrayDtoUnmarshaller instead of GitHubAllRepositoriesUnmarshaller